sdcm/scan_operation_thread.py

```diff
@@ -337,63 +482,123 @@ def _compare_output_files(self):
+            op_type=self.__class__.__name__,
+            nemesis_at_start=self.db_node.running_nemesis,
+            cmd=str(queries)
+        )
+
+        reversed_op_stat = self.run_scan_event(cmd=reversed_query, scan_event=FullPartitionScanReversedOrderEvent)
         self.reversed_query_filter_ck_stats[self.ck_filter]['count'] += 1
-        self.reversed_query_filter_ck_stats[self.ck_filter]['total_scan_duration'] += self.time_elapsed
+        self.reversed_query_filter_ck_stats[self.ck_filter]['total_scan_duration'] += self.fullscan_stats.time_elapsed
         count = self.reversed_query_filter_ck_stats[self.ck_filter]['count']
         average = self.reversed_query_filter_ck_stats[self.ck_filter]['total_scan_duration'] / count
         self.log.debug('Average %s scans duration of %s executions is: %s', self.ck_filter, count, average)
-        if self.full_partition_scan_params.get('validate_data'):
+
+        if self.fullscan_params.validate_data:
             self.log.debug('Executing the normal query: %s', normal_query)
             self.scan_event = FullPartitionScanEvent
-            super().run_scan_operation(cmd=normal_query)
-            self._compare_output_files()
+            regular_op_stat = self.run_scan_event(cmd=normal_query, scan_event=self.scan_event)
+            comparison_result = self._compare_output_files()
+            full_partition_op_stat.nemesis_at_end = self.db_node.running_nemesis
+            full_partition_op_stat.exceptions.append(regular_op_stat.exceptions)
+            full_partition_op_stat.exceptions.append(reversed_op_stat.exceptions)
+            if comparison_result and not full_partition_op_stat.exceptions:
+                full_partition_op_stat.success = True
+            else:
+                full_partition_op_stat.success = False
+
+
+class FullScanAggregatesOperation(FullscanOperationBase):
+    def __init__(self, generator, **kwargs):
+        super().__init__(generator, scan_event=FullScanAggregateEvent, **kwargs)
+
+    def randomly_form_cql_statement(self) -> str:
+        timeout = self.generator.choice(TIMEOUT_VALUES)
+        bypass_cache = self.generator.choice(BYPASS_CACHE_VALUES)
+        cmd = FullScanAggregateCommands.AGG_COUNT_ALL.base_query.substitute(
+            ks_cf=self.fullscan_params.ks_cf,
+            timeout=f" USING TIMEOUT {timeout}s" if timeout else "",
+            bypass_cache=bypass_cache
+        )
+        return cmd
+
+    def execute_query(self, session, cmd: str) -> ResultSet:
+        cmd_result = session.execute(query=cmd, trace=True)
+
+        if self._validate_fullscan_result(cmd_result):
+            self.scan_event.message = f"{type(self).__name__} operation ended successfully"
+        else:
+            self.scan_event.severity = Severity.WARNING
+            self.scan_event.message = f"{type(self).__name__} operation failed."
+
+        return cmd_result
+
+    def _validate_fullscan_result(self, cmd_result: ResultSet):
```

Comment:
how about returning a failure message only?
In case no message -> all good, operation successful
In case of message -> add this message to operation failed event

and in case of failing to find `Dispathing forward request...` also return this message so we know operation failure reason

Reply:
ok, I see, this comes from reverted commit so I cannot refer to it easily.
Generally, I was thinking of:
function `_validate_fullscan_result` in `FullScanAggregatesOperation` returns the status of operation (`False`, for failure and 'True` for success) but gives no information about the reason. When investigating, one must go into logs to check the reason.

I propose `_validate_fullscan_result` to return the reason of failure as string (empty string or None for success).
This message should be appended to the warning event.
We can consider setting `self.scan_event.severity` inside  `_validate_fullscan_result` and dropping some logic from `execute_query` - so we don't have to return anything.